### ISSUE TITLE
ErrorException in LaravelDataSource.php line 76

### ISSUE THREAD

**Issue by davidnknight** [itsgoingd/clockwork#95]
I'm throwing an exception in a Laravel app using the following:
`throw new NotFoundHttpException("{$entity} not found.");`

Which uses:
`use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;`

Which in turn results in:
```ErrorException in LaravelDataSource.php line 76:
Argument 1 passed to Clockwork\DataSource\LaravelDataSource::setResponse() must be an instance of Symfony\Component\HttpFoundation\Response, null given, called in /.../vendor/itsgoingd/clockwork/Clockwork/Support/Laravel/ClockworkSupport.php on line 70 and defined```

**Comment by itsgoingd** [comment]
Hey, what version of Laravel are you using? Can you post your HttpKernel? Works fine for me with default installation of 5.0.30.

**Comment by davidnknight** [comment]
Hey,

Tested it myself in a fresh Laravel install afterwards as I was also running the latest version of Laravel (5.0.30) and found it to work in the fresh install.

After much hunting, I narrowed the cause of the problem down to a middleware that wasn't returning the closure outside of a conditional, basically missing `return $next($request);`. It seems the request isn't immediately aborted on exception and Laravel does more with it first, and due to the lack of returning the closure, the response was null, which is what Clockwork was also complaining about, but ultimately wasn't a problem with Clockwork itself.

Apologies for misreporting and thanks for getting back to me so quickly, appreciate it. :)
